recipes/libbsc/multithreading.patch

```diff
@@ -0,0 +1,126 @@
+From 25d153b2efc4b64ef96232ded930079d0ccc394f Mon Sep 17 00:00:00 2001
+From: Shubham Chandak <schandak@stanford.edu>
+Date: Wed, 2 Aug 2017 12:35:49 -0700
+Subject: [PATCH 1/3] Adding support for no. of threads
+
+---
+ bsc.cpp | 16 ++++++++++++----
+ 1 file changed, 12 insertions(+), 4 deletions(-)
+
+diff --git a/bsc.cpp b/bsc.cpp
+index 4b80315..5523885 100644
+--- a/bsc.cpp
++++ b/bsc.cpp
+@@ -78,7 +78,7 @@ int paramBlockSorter              = LIBBSC_BLOCKSORTER_BWT;
+ int paramCoder                    = LIBBSC_CODER_QLFC_STATIC;
+ int paramSortingContexts          = LIBBSC_CONTEXTS_FOLLOWING;
+ 
+-int paramEnableParallelProcessing = 1;
++int paramEnableParallelProcessing = 0;
+ int paramEnableMultiThreading     = 1;
+ int paramEnableFastMode           = 1;
+ int paramEnableLargePages         = 0;
+@@ -88,6 +88,7 @@ int paramEnableReordering         = 0;
+ int paramEnableLZP                = 1;
+ int paramLZPHashSize              = 16;
+ int paramLZPMinLen                = 128;
++int paramnumthr			  = 1;
+ 
+ int paramFeatures()
+ {
+@@ -193,7 +194,7 @@ void Compression(char * argv[])
+     int numThreads = 1;
+     if (paramEnableParallelProcessing)
+     {
+-        numThreads = omp_get_max_threads();
++        numThreads = paramnumthr;
+         if (numThreads <= nBlocks) paramEnableMultiThreading = 0;
+         if (numThreads >= nBlocks) numThreads = nBlocks;
+     }
+@@ -717,7 +718,7 @@ void ShowUsage(void)
+     fprintf(stdout, "  -P       Enable large 2MB RAM pages, default: disable\n");
+ #endif
+ #ifdef LIBBSC_OPENMP
+-    fprintf(stdout, "  -t       Disable parallel blocks processing, default: enable\n");
++    fprintf(stdout, "  -t<num_thr>       Number of threads: default 1\n");
```

Comment:
This is a breaking change. Can you make the default to be `omp_get_max_threads()`?

Reply:
I have made another commit removing the patch. I realized that I was not using the features of the patch in this project, so it's probably better to just go with the original.